application/lib/model/Task.dart

```diff
@@ -11,32 +15,67 @@ enum Status {
 enum Priority { low, medium, high, critical }
 
 class Task implements Json {
-  Task({this.taskName, this.appId, this.screenId, this.taskDescription = ''});
+  Task({this.taskName, this.appId, this.screenId, this.taskDescription = '', this.application});
 
   @override
-  Task.fromJson(Map<String, dynamic> json) {
-    taskName = json['taskName'];
+  Task.fromJson(this.application, Map<String, dynamic> json) {
+    id = json['id'];
+    taskName = json['name'];
     appId = json['appId'];
     taskDescription = json['description'];
     screenId = <int>[];
     json['screenId'].forEach((dynamic element) {
       screenId.add(element);
     });
+    issueUrl = json['issueUrl'];
   }
-
+  Application application;
   String taskName;
   Status taskStatus;
   String taskDescription;
   int appId;
+  int id;
   Priority taskPriority = Priority.low;
   bool newInformation = false;
   List<int> screenId;
-
-  // TODO(ALL): Implement github url, depends on https://github.com/SW814F20/Application/issues/37
-  // TODO(ALL): Implement github url https://github.com/SW814F20/Application/issues/38
+  Issue githubIssue;
   String issueUrl;
 
-  // TODO(ALL): Implement github api, https://github.com/SW814F20/Application/issues/36
+  Future<Task> getGithubInformation() async {
+    githubIssue = await application.githubApi.getIssue(issueUrl);
+    setGithubInformation(githubIssue);
+    return this;
+  }
+
+  void setGithubInformation(Issue issue) {
+    taskStatus = getStatusFromGithubIssue(issue);
+    taskPriority = getPriorityFromGithubIssue(issue);
+    githubIssue = issue;
+  }
+
+  Status getStatusFromGithubIssue(Issue issue) {
+    Status status;
+    for (Label label in issue.labels) {
+      try {
+        status = BaseApi.convertStringToStatus(label.name);
+      } catch (exception) {
+        // Ignore
+      }
```

Comment:
That would not give the same result. This way if the label cannot be converted to a status, it is skipped, and status is not set.

Reply:
Well since you have the uninitialized variable `status` and in dart uninitialized variables have a default value of null, the behavior would be equal would it not ?

Reply:
Yes, but this is in a loop, if status is set to something different to null, and then later is set to null, the final status will be null, and not the value set.

The value might not be the first or the last